packages/aws-cdk/lib/import.ts

```diff
@@ -153,6 +153,10 @@ export class ResourceImporter {
         resourcesToImport,
       });
 
+      if (result.type !== 'did-deploy-stack') {
```

Comment:
This is where const/enum will shine. If there is a typo on the RHS, the check will do unexpected things.

Reply:
It's already typed. If there's a typo in the RHS, it will not compile.